### PR TITLE
Cleanup articles cache busting

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### UI accessibility concerns?

None, this is a backend service.

## Added tests?

- [ ] Yes
- [x] No, and this is why: Refactor - we do have tests around organization slug - and the exact count of times we call the cache buster - so that hasn't changed. I don't know if we have test coverage around article.collection but that should be unchanged.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: Refactor

## [optional] What gif best describes this PR or how it makes you feel?

image

### CODE REVIEW

**Comment by vaidehijoshi** [comment]
(Just leaving a note that I requested @atsmith813's review here as I know he has worked on this code fairly recently! 😃)

File: app/services/edge_cache/bust_article.rb [hunk 1]
```diff
@@ -14,26 +14,41 @@ def self.call(article)
...
 
       return unless article.collection_id
 
       article.collection.articles.find_each do |collection_article|
-        cache_bust.call(collection_article.path)
+        yield collection_article.path
       end
     end
 
+
```
Comment (citizen428) [inline app/services/edge_cache/bust_article.rb:51]: A wild newline appears!

Comment (djuber) [reply app/services/edge_cache/bust_article.rb:51]: whitespace butler mode fails me again... thanks for catching this

**Review by atsmith813 — APPROVED** [review]
LGTM!

FWIW we can also pass the array of paths directly to `.call` instead of looping and calling `call` on each one if you prefer that route.

I suppose that would essentially result in looping twice, but we do it here and there.
https://github.com/forem/forem/blob/29dcea37503d903775c592f14b7dcf66d04a793a/app/services/edge_cache/bust.rb#L11-L24

https://github.com/forem/forem/blob/29dcea37503d903775c592f14b7dcf66d04a793a/app/controllers/admin/tools_controller.rb#L42-L55

**Comment by djuber** [comment]
> FWIW we can also pass the array of paths directly to .call instead of looping and calling call on each one if you prefer that route.

ah - that makes sense - Array.wrap permits either an element or an array... thanks for pointing that out!

**Review by citizen428 — APPROVED** [review]

**Pull request merged**